app/serializers/topic_view_serializer.rb

```diff
@@ -190,11 +195,53 @@ def include_expandable_first_post?
   end
 
   def bookmarked
-    object.has_bookmarks?
+    @bookmarked ||= object.has_bookmarks?
   end
 
-  def bookmarked_posts
-    object.bookmarked_posts
+  def bookmark_id
+    return if !bookmarked
+    object.bookmark&.id
```

Comment:
Will `bookmark` be blank if we're already checked that `bookmarked` is true?

Reply:
Good point!